shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1644,6 +1644,88 @@ public int getCode() {
         }
     }
 
+    /**
+     * A generic error response that encapsulates an error code (to be used for client-side processing) and an error message
+     * describing the origin of the error. This should be used to describe general exceptions where limited information is required.
+     */
+    @Data
+    public static final class ErrorMessage implements Reply, WireCommand {
+        final WireCommandType type = WireCommandType.ERROR_MESSAGE;
+        final long requestId;
+        final String message;
+        final ErrorCode errorCode;
+
+        @Override
+        public void process(ReplyProcessor cp) throws UnsupportedOperationException {
+            cp.errorMessage(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(message == null ? "" : message);
+            out.writeInt(errorCode.getCode());
+        }
+
+        public static WireCommand readFrom(EnhancedByteBufInputStream in, int length) throws IOException {
+            return new ErrorMessage(in.readLong(), in.readUTF(), ErrorCode.valueOf(in.readInt()));
+        }
+
+        public RuntimeException getThrowableException() {
```

Comment:
Looks like this is not covered by unit tests. Please add unit tests for this class.